ReportingServicesTools/Functions/Utilities/Connect-RsReportServer.ps1

```diff
@@ -32,6 +32,11 @@ function Connect-RsReportServer
         .PARAMETER ReportPortalUri
             The Uri to connect to for accessing the REST Endpoint. This exists in SQL Server Reporting Services 2016 and later.
         
+        .PARAMETER APIVersion
```

Comment:
It is a bit strange that this parameter is added to this cmdlet because it is not really doing anything. Every single parameter to this cmdlet modifies Microsoft.ReportingServicesTools.ConnectionHost (which is defined in <root>\ReportingServicesTools\Libraries\library.ps1). I recommend either removing this parameter altogether or adding ApiVersion in ConnectionHost. My only concern with adding something like ApiVersion in ConnectionHost is there are 2 ways to communicate with Reporting Services starting from 2016: SoapEndpoint found at ReportServerUri and RestEndpoint found at PortalUri (which is open to developers starting from SSRS 2017). If you do add ApiVersion to ConnectionHost, I highly recommend you to call it SoapEndpointApiVersion, so that people don't confuse it with the ApiVersion of RestEndpoint.

Reply:
I think its needed as a parameter to pass through to New-RsWebServiceProxy (although maybe this could be New-RsWebServiceProxyHelper instead?). Anyway i've gone with the solution of renaming it to $SoapEndpointApiVersion as suggested.